CRM/Upgrade/Incremental/php/FourFive.php

```diff
@@ -303,6 +303,57 @@ static function migrateHonoreeInfo(CRM_Queue_TaskContext $ctx) {
   }
 
   /**
+   * Upgrade function.
+   *
+   * @param string $rev
+   */
+  public function upgrade_4_5_6($rev) {
+    // Task to process sql.
+    $this->addTask(ts('Upgrade DB to 4.5.6: Fix saved searches consisting of multi-choice custom field(s)'), 'updateSavedSearch');
+
+    return TRUE;
+  }
+
+  /**
+   * Update saved search for multi-select custom fields on DB upgrade
+   *
+   * @param CRM_Queue_TaskContext $ctx
+   *
+   * @return bool TRUE for success
+   */
+  static function updateSavedSearch(CRM_Queue_TaskContext $ctx) {
+    $sql = "SELECT id, form_values FROM civicrm_saved_search";
+    $dao = CRM_Core_DAO::executeQuery($sql);
+    while ($dao->fetch()) {
+      $formValues = unserialize($dao->form_values);
+      while (list($field, $data_value) = each($formValues)) {
+        if (preg_match('/^custom_/', $field) && is_array($data_value)) {
```

Comment:
Monish - the difference might be the type of field. I think checkboxes and multi-selects have a different data structure. Try with both.

My main point was that your condition `preg_match('/^custom_/', $field) && is_array($data_value)` is not specific to any data structure - they all use an array, so you your code would "upgrade" fields that are already using the new data structure and change all of their operators to 'and' (which is the default you set on line 332).